---
title: "Homework 8"
author: "Chandrima Bhattacharya"
date: "7 March 2019"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r loadlib}
suppressPackageStartupMessages(library(DESeq2))
setwd("F:/Annie/CornellMS/Semester2/ANGSD/ANGSD/")
```

## Question 1

Read count defines the number of reads or fragments overlapping with the union of exons of a gene.  
Properties of read counts are influenced by the follow.  

* They are influenced by the sequencing depth. Usually, deeper sequencing gives higher expression value.
* The length of the gene and GC-bias associated with them are also responsible for difference in read counts value.
* The RNA-composition of the different individual gene abundances are responsible for affecting read counts. Highly-expressed transcripts usually can sometimes dominate reducing the reads count available for the rest of the transcripts.  

Correction techniques for the following is defined below.

* Sequencing depth needs to be corrected when we are comparing the same gene between different samples. If we are using DESeq2, we can use one of it's function *estimateSizeFactors()* to correct the depth. It uses multiple statistical normalization steps for the following. There are other functions which are defined also for the same!
* The gene length and the GC% in the transcript sequence needs to be corrected while comparing different genes. For this we can use TPM or Transcript per Million normalization technique. TPM normalizes for gene length first, and then normalize for sequencing depth second, and hence performs better than RPKM.
* Individual gene abundances needs to be corrected when we are comparing the same gene between different samples. *rlog* is a DESeq2 function which does that by transforming the the count data to the log2 scale in a way which minimizes differences between samples for rows with small counts, and which normalizes with respect to library size. 

## Question 2

```{helpData}
?rlog()
?DESeqDataSetFromMatrix
```

The input format  for rlog() is given below.

```{out1}
rlog(object, blind = TRUE, intercept, betaPriorVar, fitType = "parametric")
```

The input format for DESeqDataSetFromMatrix() is given below.

```{out2}
DESeqDataSetFromMatrix(countData, colData, design, tidy = FALSE,
  ignoreRank = FALSE, ...)
```

To view S4 objects, we can use showClass as follows.

```{r usingShowClass}
DESeq.rlog <- readRDS('rlogData.rds')
DESeqDS <- readRDS('DESeq.rds')
showClass(class (DESeq.rlog))
showClass(class (DESeqDS))
```

#### a

DESeqDataSetFromMatrix() helps perform transformations and exploratory data analysis. The rlog() transform uses a an empirical Bayesian prior on inter-sample differences in the form of a ridge penalty. It uses log2scale. The DESeqDataSetFromMatrix creates a DESeq object from a raw counts matrix and colData. The rlog() function transforms the count data.  

Similarities:  
* Both are S4 objects. They also have associated slots.
* They are DESeq2 function and thus creates DESeq objects as output.
  
Differences:
* rlog() uses a logarithmic transformation whereas DESeqDataSetFromMatrix() doesn't. 
* The function rlog() returns a SummarizedExperiment object which contains the rlog-transformed values in its assay slot. Also it needs SummarizedExperiment object as input. DESeqDataSetFromMatrix() can be used in cases where the data not in the form of a SummarizedExperiment object, but either as a simple matrix of count values or as output files.

#### b

The assay function will help us extract the expression values stored in either object.  
The function rlog returns a SummarizedExperiment object which contains the rlog-transformed values in its assay slot.

```{r}
head(assay(DESeq.rlog))
head(assay(DESeqDS))
```

#### c

I have added a new matrix to my DESeq object under the metadata slot. Now my_normalization can be called.

```{r }
DESeq.rlog@metadata$my_normalization <- assay(DESeq.rlog)
```

#### Question 3

```{r plotting}
# Dendrogram
rlog.norm.counts <- readRDS("rlog.norm.counts.rds")
pearson_cor <- cor(rlog.norm.counts, method = "pearson")
distance.m_rlog <- as.dist(1-pearson_cor)
plot(hclust(distance.m_rlog),labels=colnames(rlog.norm.counts),main="rlog transformed read counts")

# PCA plot
plotPCA(DESeq.rlog, intgroup=c("condition"))
```
Both the dendrogram and PCA shows clear data trend. We see that the knock outs cluster together as they are similar gene-knockouts. The wild-type is seen in the different side of the graph but are more spread out. We can see clustering of wild type 2,5 and 3,4 which is represented also by dendogram. The WT and KO conditions are definitely distinct from each other which is why they are spread out so distantly. This is confirmed in the sample distance heatmap in the pcaExplorer. 

PCA Explorer allows to view  top and bottom loadings plot. I think the various Principal Components which get formed while PCA is shown in the diagram. 

```{}
#PCA Explorer
pcaExplorer::pcaExplorer(dds=DESeqDS ,dst=DESeq.rlog)
```
